code/modules/mob/living/living.dm

```diff
@@ -917,9 +917,15 @@ Thanks.
 	//putting out a fire
 		if(CM.on_fire && CM.canmove)
 			CM.fire_stacks -= 5
-			CM.weakened = 5
+			CM.SetWeakened(3)
+			playsound(CM.loc, 'sound/effects/bodyfall.ogg', 50, 1)
 			CM.visible_message("<span class='danger'>[CM] rolls on the floor, trying to put themselves out!</span>",
 							   "<span class='warning'>You stop, drop, and roll!</span>")
+
+			for(var/i in list(1, 4, 2, 8, 1, 4, 2, 8, 1, 4))
```

Comment:
@9600bauds it's so they can be used as bitflags to generate the diagonals
NORTH and EAST is 1 and 4, 5.
SOUTH and EAST is 2 and 4, 6.

and so on.
Directions being bits allows for a lot of common operations based on direction to be fast.
they also allow you to store multiple directions in one variable easily.